lib/charms/sdcore_nms_k8s/v0/fiveg_core_gnb.py

```diff
@@ -0,0 +1,348 @@
+            )
+            ...
+            # on relation-joined the charm shall publish the CU/gNodeB name to the databag
+            self.framework.observe(
+                self.on.fiveg_core_gnb_relation_joined, self._on_fiveg_core_gnb_relation_joined
+            )
+            self.framework.observe(self.on.fiveg_core_gnb_relation_changed,
+                self._on_fiveg_core_gnb_relation_changed)
+
+        def _on_fiveg_core_gnb_relation_joined(self, event: RelationJoinedEvent):
+            relation_id = event.relation.id
+            self.fiveg_core_gnb.publish_gnb_information(
+                relation_id=relation_id,
+                cu_name=self.CU_NAME,
+            )
+
+        def _on_fiveg_core_gnb_relation_changed(self, event: RelationChangedEvent):
+            tac = event.tac,
+            plmns = event.plmns,
+            # Do something with the TAC and PLMNs.
+    ```
+
+    And a corresponding section in charm's `charmcraft.yaml`:
+    ```
+    requires:
+        fiveg_core_gnb:  # Relation name
+            interface: fiveg_core_gnb  # Relation interface
+    ```
+"""
+import json
+import logging
+from dataclasses import dataclass
+from typing import Optional
+
+from interface_tester.schema_base import DataBagSchema
+from ops.charm import CharmBase
+from ops.framework import Object
+from pydantic import BaseModel, Field, ValidationError, conlist
+
+# The unique Charmhub library identifier, never change it
+LIBID = "196ff8f539ba4f2998209fbb50e2dbbf"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+logger = logging.getLogger(__name__)
+
+"""Schemas definition for the provider and requirer sides of the `fiveg_core_gnb` interface.
+It exposes two interfaces.schema_base.DataBagSchema subclasses called:
+- ProviderSchema
+- RequirerSchema
+
+Examples:
+    ProviderSchema:
+        unit: <empty>
+        app: {
+            "tac": 1,
+            "plmns": [
+                {
+                    "mcc": "001",
+                    "mnc": "01",
+                    "sst": 1,
+                    "sd": 1,
+                }
+            ],
+        }
+    RequirerSchema:
+        unit: <empty>
+        app: {
+            "cu_name": "gnb001",
+        }
+"""
+
+
+@dataclass
+class PLMNConfig(BaseModel):
+    """Dataclass representing the configuration for a PLMN."""
+
+    def __init__(self, mcc: str, mnc: str, sst: int, sd: Optional[int] = None) -> None:
+        super().__init__(mcc=mcc, mnc=mnc, sst=sst, sd=sd)
+
+    mcc: str = Field(
+        description="Mobile Country Code",
+        examples=["001", "208", "302"],
+        pattern=r"[0-9][0-9][0-9]",
+    )
+    mnc: str = Field(
+        description="Mobile Network Code",
+        examples=["01", "001", "999"],
+        pattern=r"[0-9][0-9][0-9]?",
```

Comment:
```suggestion
        pattern=r"^[0-9][0-9][0-9]?$",
```

Reply:
otherwise we can have 3 or more numbers. To be updated in the interface repo as well